hbase-server/src/main/java/org/apache/hadoop/hbase/master/balancer/StochasticLoadBalancer.java

```diff
@@ -348,9 +348,10 @@ protected boolean needsBalance(Cluster cluster) {
     if (total <= 0 || sumMultiplier <= 0
         || (sumMultiplier > 0 && (total / sumMultiplier) < minCostNeedBalance)) {
       if (LOG.isTraceEnabled()) {
-        LOG.trace("Skipping load balancing because balanced cluster; " + "total cost is " + total
-          + ", sum multiplier is " + sumMultiplier + " min cost which need balance is "
-          + minCostNeedBalance);
+        final String loadBalanceTarget = isByTable ? String.format("table (%s)", tableName) : "cluster";
```

Comment:
nit: although loadBalanceTarget is used just for logging, it is better to keep only log.trace() within LOG.isTraceEnabled()

Reply:
@virajjasani better to keep like this, as we don't need the extra execution in case TRACE isn't on, and keeping it as a separate variable gives better readability.

Reply:
Definitely keeping it as a variable is fine. I just thought it would be better to keep it out of  `LOG.isTraceEnabled()` since usually we want only `LOG.trace()` to be covered by it. But anyways, keeping it unchanged is fine, it's just used by trace log.